### DISCUSSION TITLE
Anyone got opencanary to work on Raspberry Pi in 2024?

### DISCUSSION THREAD

**Discussion by bear-goes-bleep** [thinkst/opencanary#366] [General]
Hey all, 

There are a few guides on how to run Opencanary on Raspberry PI, but they all seem dated. Can't even get it to start by following any of these guides. Various issues, from missing dependencies and links to twistd I can't solve. (not super proficient in Linux...)  

Is RPI a no go here? 

Thanks

**Comment by theidiotyouyellat** [comment]
@bear-goes-bleep I've had multiple running on raspberrypi's using Ubuntu 22.04 and ansible to configure opencanary.

**Reply by bear-goes-bleep** [reply]
did you install any of them recently? Most guides I have seen are from 2022 or earlier.

**Reply by theidiotyouyellat** [reply]
I have installed them as recently as March 2024 on RaspberryPi 4's using [this](https://github.com/thinkst/ansible-role-opencanary) on Ubuntu 22.04.

**Reply by bear-goes-bleep** [reply]
Ubuntu **22**.04 LTS is the key here! 

Documentation states 20.04 LTS so I have built all my RPI's on that. While troubleshooting late last night I realised that 20.04 runs python 3.8 or something, but python 3.9 is required. Did install 3.9 yesterday but it still did not work, probably a messed up installation/virtualenv. 

Rebuilt rpi with Ubuntu 22.04 instead, tested just now and Opencanary installed perfectly according to instructions! 👍 

Will report that they update install doc. 

Thanks! :)

**Reply by theidiotyouyellat** [reply]
Nice! Glad that worked for you. Cheers
